service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/RoutingGatewayFilterFactory.java

```diff
@@ -49,6 +54,11 @@ public RouteLocator routes(RouteLocatorBuilder builder) {
         return builder.routes()
             .route("get-document", r -> r.path(documentPath)
                 .and()
+                .header(INTERACTION_ID_HEADER_NAME, DOCUMENT_READ_ID)
+                .uri(getDefaultTargetUri()))
+            .route("migrate-document", r -> r.path(documentPath)
+                .and()
+                .header(INTERACTION_ID_HEADER_NAME, DOCUMENT_MIGRATE_ID)
                 .uri(getDefaultTargetUri()))
```

Comment:
Including header-based predicates here, because these two endpoints have the same path and the only difference is the `Ssp-InteractionID` header vvalue.